Kbs.Wpf/User/Update/AccountView.xaml.cs

```diff
@@ -0,0 +1,152 @@
+                {
+                    Email = ViewModel.InputEmail,
+                    Password = ViewModel.InputPassword
+                };
+                var validationResult = _userValidator.ValidatorForUpdate(user, PasswordConfirmInput.Password, _userRepository);
+                if (user.Email != null && user.Email.Length != 0 && !user.Email.Equals(SessionManager.Instance.Current.User.Email))
+                {
+                    mailChange = true;
+                    if (validationResult.TryGetValue(nameof(user.Email), out string emailMessage))
+                    {
+                        EmailErrorMessage.Content = emailMessage;
+                        EmailErrorMessage.Visibility = Visibility.Visible;
+                    }
+                    else
+                    {
+                        EmailErrorMessage.Visibility = Visibility.Hidden;
+                        mailCorrect = true;
+                    }
+                } 
+
+                if (user.Password != null && user.Password.Length != 0)
+                {
+                    passwordChange = true;
+                    if (validationResult.TryGetValue(nameof(user.Password), out string passwordMessage))
+                    {
+                        PasswordErrorMessage.Text = passwordMessage;
+                        PasswordErrorMessage.Visibility = Visibility.Visible;
+                    } else if (validationResult.ContainsKey("Bevestiging"))
+                    {
+                        PasswordErrorMessage.Visibility = Visibility.Hidden;
+                        matching = false;
+                    }
+                    else
+                    {
+                        PasswordErrorMessage.Visibility = Visibility.Hidden;
+                        passwordCorrect = true;
+                    }
+                }
+
+                if (validationResult.Count > 1 || !matching)
+                {
+                    return;
+                }
+
+                bool different = false;
+                if (mailChange && mailCorrect)
+                {
+
+                    if (!user.Email.Equals(SessionManager.Instance.Current.User.Email))
+                    {
+                        different = true;
+                        SessionManager.Instance.Current.User.Email = user.Email;
+                    } else
+                    {
+                        mailChange = false;
+                    }
+                }
+                if (passwordChange && passwordCorrect)
+                {
+                    user.Encrypt();
+                    if (!user.Password.Equals(SessionManager.Instance.Current.User.Password))
+                    {
+                        different = true;
+                        SessionManager.Instance.Current.User.Password = user.Password;
+                    }
+                }
+                if (different)
+                {
+                    UserRepository repository = new UserRepository();
```

Comment:
you shouldn't need to initiate a new UserRepository here, use a local variable instead